### ISSUE TITLE
Video blank black screen after seek ( link included )

### ISSUE THREAD

**Issue by DannyZB** [mangui/flashls#419]
URL:https://nl3.seedr.co.il:444/media/video/hd_hls/342881110017bda34923b2e4071d2385583a65b4/vid.m3u8

With chromeless player : http://www.flashls.org/latest/examples/chromeless/?src=https%3A%2F%2Fnl3.seedr.co.il%3A444%2Fmedia%2Fvideo%2Fhd_hls%2F342881110017bda34923b2e4071d2385583a65b4%2Fvid.m3u8

see for yourself

Seeking to some parts of the video works well , while for others the issue happens.
The video works perfectly well on the dailymotion HTML5 HLS player, so we know the stream is valid.

This was the issue I thought was related to keyframe-mismatching between qualities, but this one has the issue on it's own.

**Comment by mangui** [comment]
if by any chance you can tell me whether it is a regression or not , by checking older releases, it would speed up the investigation.
thanks

**Comment by DannyZB** [comment]
As far as I know it's not a regression.
Used an older version until recently with the same issue ( my earlier pull was from half a year ago I believe )

p.s. I had to move the file to a different server

It's now at : http://www.flashls.org/latest/examples/chromeless/?src=https%3A%2F%2Fnl3.seedr.co.il%3A444%2Fmedia%2Fvideo%2Fhd_hls%2F342881110017bda34923b2e4071d2385583a65b4%2Fvid.m3u8

https://nl3.seedr.co.il:444/media/video/hd_hls/342881110017bda34923b2e4071d2385583a65b4/vid.m3u8

**Comment by mangui** [comment]
I checked your stream, first fragment contains everything needed to initialize the video decoder : 
SPS,PPS = sequence parameter set, picture parameter set : http://www.cardinalpeak.com/blog/the-h-264-sequence-parameter-set
IDR = keyframe

```
INFO:AVC: AUD,SEI,SEI,SPS,PPS,IDR slices
```

but when you seek to another fragment,only IDR is found.

```
INFO:HLSNetStream:seek(2747.1391676431767)
INFO:AVC: AUD,SEI,IDR slices
```

you won't find any SPS,PPS anymore => the video decoder cannot be initialized.

hls.js is able to decode it by luck, because it keeps SPS,PPS in memory (although in theory it should not), but IMHO your stream is not compliant. all fragments should be independent and self decodable without requiring info from previous fragments.

**Comment by DannyZB** [comment]
Encoding that way saves massive computing power, 
as you can see modern HLS encoders/decoders have no issues with it.
Can't it be supported ?

HLS is changing rapidly, haven't ever seen a single implementation up to the theoretical spec - not even the one by apple.

Supporting this makes for much lighter streams.

This is what ffmpeg produces if I don't force it to generate keyframes at specific intervals ( which sometimes forces re-encoding entire segments thus losing quality )

**Comment by mangui** [comment]
true, there are lots of varations and encoding flavors around HLS and other fragmented video delivery mechanisms ... but usually all of them share the same key principle :  having independently decodable fragments.

**Comment by DannyZB** [comment]
https://ffmpeg.org/pipermail/ffmpeg-user/2014-June/021836.html

For reference,
if anyone runs into this later with the ffmpeg segmenter.

Seems ffmpeg has a serious issue with its segmenter , or more accurately with unreadable online documentation.

As it seems, newer implementations of the HLS segmenter segment at keyframes automatically, negating the need for re-rendering on the stream.
The issue here is that when using "-vcodec copy" and the ssegment segmenter it doesnt add PPS/SPS to each segment generated.

Adding -bsf:v +dump_extra seems to have solved that for now, 
hopefully I got that part right.
